src/learning/InitializeProfile.cpp

```diff
@@ -0,0 +1,203 @@
+#include "../../extsrc/spdlog/include/spdlog/fmt/ostr.h"
+#include "../../extsrc/spdlog/include/spdlog/spdlog.h"
+
+#include "../../include/learning/InitializeProfile.h"
+#include "../../include/types/AlternativesPerformance.h"
+#include "../../include/types/Categories.h"
+#include "../../include/types/Criteria.h"
+#include "../../include/types/MRSortModel.h"
+#include "../../include/utils.h"
+
+#include <algorithm>
+#include <iostream>
+#include <numeric>
+#include <string>
+#include <typeinfo>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+ProfileInitializer::ProfileInitializer(Config &config,
+                                       AlternativesPerformance &altPerfs)
+    : conf(config), altPerformance_(altPerfs) {
+  conf.logger->debug("Starting ProfileInitializer object...");
+}
+
+ProfileInitializer::ProfileInitializer(Config &config,
+                                       const ProfileInitializer &profInit)
+    : conf(config), altPerformance_(profInit.getAlternativesPerformance()) {
+  conf.logger->debug("Starting ProfileInitializer object...");
+}
+
+AlternativesPerformance ProfileInitializer::getAlternativesPerformance() const {
+  return altPerformance_;
+}
+
+void ProfileInitializer::setAlternativesPerformance(
+    AlternativesPerformance &newAltPerfs) {
+  altPerformance_ = newAltPerfs;
+}
+
+std::vector<float> ProfileInitializer::categoryFrequency() {
+
+  // Extraction category rank from each alternative and storing it in values
+  std::unordered_map<std::string, Category> map =
+      altPerformance_.getAlternativesAssignments();
+  std::vector<int> values(map.size());
+
+  auto cat_value_selector = [](auto pair) {
+    return pair.second.getCategoryRank();
+  };
+
+  std::transform(map.begin(), map.end(), values.begin(), cat_value_selector);
+
+  // Counting the number of occurence for each category rank
+  std::unordered_map<int, int> counts;
+  for (auto v : values)
+    ++counts[v];
+
+  // Counting the number of alternatives in the altPerformance object
+  const int nbAlternatives = std::accumulate(
+      std::begin(counts), std::end(counts), 0,
+      [](int value, const std::unordered_map<int, int>::value_type &p) {
+        return value + p.second;
+      });
+
+  // Computing the frequency
+  std::vector<float> frequency;
+  for (auto const pair : counts) {
+    float c = pair.second;
+    frequency.push_back(c / nbAlternatives);
+  }
+  return frequency;
+}
+
+std::vector<std::string> ProfileInitializer::getProfilePerformanceCandidates(
```

Comment:
In general this function could be implemented in a fastest way, the for loop for only one profile is ineficient: we could go over the alternativesAssignments only once and compute all the candidates for each profile at the same time instead of going over the aa n_profiles time